### ISSUE TITLE
#track 2: can you release the upper-bound performance of task1-task5? The test set contains 277 categories, while the training set of task1 only has 159 categories, then the upper-bound of task1 is?

### ISSUE THREAD

**Issue by Layjins** [ContinualAI/clvision-challenge-2022#12]

**Comment by lrzpellegrini** [comment]
Hi @Layjins,

the maximum mAP on the test set that can be obtained after each task is:

After task 0: `0.57400722`
After task 1: `0.76173285`
After task 2: `0.86642599`
After task 3: `0.93862816`
After task 4: `1.00000000`

Those maximum scores are obtained as: `len(categories_seen_so_far) / overall_categories`.

This means that the overall upper bound is `0.828158844765343`, which is the mean of these values.

This is the detailed breakdown of categories encountered in each experience:

Experience 0 features 159 categories
of which never seen before: 159
of which already seen: 0
already seen, but not in this experience: 0
for a total of seen so far (past+now): 159

Experience 1 features 147 categories
of which never seen before: 52
of which already seen: 95
already seen, but not in this experience: 64
for a total of seen so far (past+now): 211

Experience 2 features 149 categories
of which never seen before: 29
of which already seen: 120
already seen, but not in this experience: 91
for a total of seen so far (past+now): 240

Experience 3 features 160 categories
of which never seen before: 20
of which already seen: 140
already seen, but not in this experience: 100
for a total of seen so far (past+now): 260

Experience 4 features 152 categories
of which never seen before: 17
of which already seen: 135
already seen, but not in this experience: 125
for a total of seen so far (past+now): 277 <--- this is the total number of categories in track 2

For instance, let's consider experience 2. After training on experience 2, 240 categories have been seen (the ones encountered in exp0 + the ones in exp1 + the ones in exp2). Considering that there is no way you can return a correct prediction for the 37 categories not seen so far, and considering that the test set contains all categories (even the ones not seen so far), this means that the optimal score after experience 2 is `240 / 277 = 0.86642599`.

I'll add more info on this in the challenge page!

**Comment by Layjins** [comment]
thanks a lot!

------------------ Original ------------------
From: Lorenzo Pellegrini ***@***.***&gt;
Date: Tue, Apr 19, 2022 11:03 PM
To: ContinualAI/clvision-challenge-2022 ***@***.***&gt;
Cc: Layjuns ***@***.***&gt;, Mention ***@***.***&gt;
Subject: Re: [ContinualAI/clvision-challenge-2022] #track 2: can you releasethe upper-bound performance of task1-task5? The test set contains 277categories, while the training set of task1 only has 159 categories, then theupper-bound of task1 is? (Issue #12)





 
Hi @Layjins,
 
the maximum mAP on the test set that can be obtained after each task is:
 
After task 0: 0.57400722
 After task 1: 0.76173285
 After task 2: 0.86642599
 After task 3: 0.93862816
 After task 4: 1.00000000
 
Those maximum scores are obtained as: len(categories_seen_so_far) / overall_categories.
 
This means that the overall upper bound is 0.828158844765343, which is the mean of these values.
 
This is the detailed breakdown of categories encountered in each experience:
 
Experience 0 features 159 categories
 of which never seen before: 159
 of which already seen: 0
 already seen, but not in this experience: 0
 for a total of seen so far (past+now): 159
 
Experience 1 features 147 categories
 of which never seen before: 52
 of which already seen: 95
 already seen, but not in this experience: 64
 for a total of seen so far (past+now): 211
 
Experience 2 features 149 categories
 of which never seen before: 29
 of which already seen: 120
 already seen, but not in this experience: 91
 for a total of seen so far (past+now): 240
 
Experience 3 features 160 categories
 of which never seen before: 20
 of which already seen: 140
 already seen, but not in this experience: 100
 for a total of seen so far (past+now): 260
 
Experience 4 features 152 categories
 of which never seen before: 17
 of which already seen: 135
 already seen, but not in this experience: 125
 for a total of seen so far (past+now): 277 <--- this is the total number of categories in track 2
 
For instance, let's consider experience 2. After training on experience 2, 240 categories have been seen (the ones encountered in exp0 + the ones in exp1 + the ones in exp2). Considering that there is no way you can return a correct prediction for the 37 categories not seen so far, and considering that the test set contains all categories (even the ones not seen so far), this means that the optimal score after experience 2 is 240 / 277 = 0.86642599.
 
I'll add more info on this in the challenge page!
